src/main/java/com/tabnine/lifecycle/EnterprisePluginInstaller.kt

```diff
@@ -0,0 +1,99 @@
+package com.tabnine.lifecycle
+
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.ide.plugins.PluginNode
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.updateSettings.impl.PluginDownloader
+import com.intellij.openapi.util.JDOMUtil
+import com.intellij.util.text.SemVer
+import com.intellij.util.xmlb.XmlSerializer
+import com.intellij.util.xmlb.annotations.Attribute
+import com.intellij.util.xmlb.annotations.Tag
+import com.tabnine.dialogs.Dialogs
+import com.tabnine.general.StaticConfig
+import com.tabnine.general.StaticConfig.TABNINE_ENTERPRISE_ID_RAW
+import com.tabnine.general.Utils
+import java.net.URL
+import java.util.concurrent.locks.ReentrantLock
+
+data class Plugin(
+    @Tag("name") val name: String? = null,
+    @Attribute("id") var id: String? = null,
+    @Attribute("version") private var version: String? = null,
+    @Attribute("url") var url: String? = null
+) {
+    val parsedVersion: SemVer?
+        get() = SemVer.parseFromText(version)
+}
+
+class TabnineEnterprisePluginInstaller {
+    private val downloadLock = ReentrantLock()
+    fun installTabnineEnterprisePlugin() {
+        val host = StaticConfig.getTabnineEnterpriseHost()
+        if (!host.isPresent) return
+        val plugin = getPluginDescriptor(host.get()) ?: return
+
+        ProgressManager.getInstance().run(object : Task.Modal(null, "Downloading Tabnine Enterprise Plugin", true) {
+            override fun run(indicator: ProgressIndicator) {
+                try {
+                    Utils.criticalSection(downloadLock) {
+                        downloadAndInstall(indicator, plugin)
+                    }
+                } catch (e: Throwable) {
+                    Logger.getInstance(javaClass).warn("Failed to install Tabnine Enterprise plugin", e)
+                }
+            }
+        })
+    }
+
+    private fun getPluginDescriptor(host: String): Plugin? {
```

Comment:
```suggestion
    private fun getTabninePluginDescriptor(host: String): Plugin? {
```
just so it'll be clear that its from our repo and not some GA marketplace